base/system/winlogon/notify.c

```diff
@@ -356,23 +357,31 @@ CallNotificationDll(
                                    &dwType,
                                    (PBYTE)szFuncBuffer,
                                    &dwSize);
+
+        RegCloseKey(hDllKey);
     }
 
-    if (dwError == ERROR_SUCCESS)
-    {
-        hModule = LoadLibraryW(NotificationDll->pszDllName);
-        if (hModule != NULL)
-        {
-            pNotifyHandler = (PWLX_NOTIFY_HANDLER)GetProcAddress(hModule, szFuncBuffer);
-            if (pNotifyHandler != NULL)
-                pNotifyHandler(pInfo);
+    if (dwError != ERROR_SUCCESS)
+        return;
 
-            FreeLibrary(hModule);
-        }
+    hModule = LoadLibraryW(NotificationDll->pszDllName);
+    if (!hModule)
+        return;
+
+    _SEH2_TRY
+    {
+        pNotifyHandler = (PWLX_NOTIFY_HANDLER)GetProcAddress(hModule, szFuncBuffer);
```

Comment:
nit: I would move this outside the SEH block.

Reply:
Moved :+1: